server/server.go

```diff
@@ -153,6 +155,20 @@ type Metrics interface {
 	InstrumentHandler(handler http.Handler, label string) http.Handler
 }
 
+// TLSConfig represents the TLS configuration for the server.
+// This configuration is used to configure file watchers to reload each file as it
+// changes on disk.
+type TLSConfig struct {
+	// CertFile is the path to the certificate file.
+	CertFile string
+
+	// KeyFile is the path to the key file.
+	KeyFile string
+
+	// CertPoolFile is the path to the CA cert pool file.
+	CertPoolFile string
```

Comment:
This goes in line with existing naming conventions, but it'd be nice if we were a bit more clear about the use of these fields; like explicitly saying the cert and key relates to the server cert, and the cert pool relates to trusted client CAs.